chalk-integration/src/program.rs

```diff
@@ -326,6 +327,30 @@ impl tls::DebugContext for Program {
     }
 }
 
+impl UnificationDatabase<ChalkIr> for Program {
```

Comment:
I should have these actually use something we can test. But I'm not actually sure the best way to approach this.

Reply:
I think we want to add explicit annotations probably into the chalk syntax, rather than reproduce rustc's inference.